src/composables/notifications.ts

```diff
@@ -0,0 +1,155 @@
+import { computed, ref, watch } from '@vue/composition-api';
+import {
+  IAccount,
+  INetwork,
+  INotification,
+  INotificationSetting,
+  NotificationStatus,
+} from '../types';
+import { useGetter, useState } from './vuex';
+import {
+  NOTIFICATION_STATUS_CREATED,
+  NOTIFICATION_STATUS_READ,
+  NOTIFICATION_TYPE_WALLET,
+  executeAndSetInterval,
+  watchUntilTruthy,
+} from '../popup/utils';
+import { getAllNotifications, modifyNotifications, normalizeNotification } from '../popup/utils/notifications';
+
+const CHUNK_SIZE = 20;
+
+const superheroNotifications = ref<INotification[]>([]);
+const walletNotifications = ref<INotification[]>([]);
+let pollingInterval: NodeJS.Timer | null = null;
+
+export function useNotifications(notificationSettings: INotificationSetting[]) {
+  const sdk = useGetter('sdkPlugin/sdk');
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+  const account = useGetter<IAccount>('account');
+
+  const canLoadMore = ref(true);
+  const offset = ref(0);
+
+  const notificationSettingsCheckedTypes = computed<string[]>(
+    () => notificationSettings.filter((setting) => setting.checked).map((s) => s.type),
+  );
+
+  const filteredNotifications = computed<INotification[]>(
+    () => [...superheroNotifications.value, ...walletNotifications.value]
+      .filter(
+        ({ status, type }) => (
+          status === NOTIFICATION_STATUS_READ
+          || notificationSettingsCheckedTypes.value.includes(type)
+        ),
+      )
+      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
+  );
+
+  const notificationsToShow = computed<INotification[]>(
+    () => filteredNotifications.value.slice(0, offset.value * CHUNK_SIZE),
+  );
+
+  async function getSuperheroNotifications() {
```

Comment:
in other composables we used arrow function

Reply:
I think in all of the components (and composables as well) we try to use 'function'